### PR TITLE
[Find forms] Accessibility tweaks

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR does some 508 improvements to Find Forms.

Another PR for https://github.com/department-of-veterans-affairs/va.gov-team/issues/7096

## Testing done
Inspected page locally

## Screenshots
N/A

## Acceptance criteria
- [ ] 508 improvements implemented

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/components/FindVaForms.jsx [hunk 1]
```diff
@@ -26,6 +26,9 @@ export default function() {
           >
             <span>
               Read more
+              <span className="vads-u-visibility--screen-reader">
+                about filing a VA disability claim
+              </span>
```
Comment (ncksllvn) [inline src/applications/find-forms/components/FindVaForms.jsx:31]: This should make the SR read `Read more about filing a VA disability claim`

File: src/applications/find-forms/containers/SearchResults.jsx [hunk 2]
```diff
@@ -118,10 +118,13 @@ export class SearchResults extends Component {
...
       <>
-        <p className="vads-u-margin-y--1p5" data-forms-focus>
+        <h2
+          className="vads-u-font-size--base vads-u-line-height--3 vads-u-font-family--sans vads-u-font-weight--normal vads-u-margin-y--1p5"
+          data-forms-focus
+        >
           Showing {startLabel} &ndash; {lastLabel} of {results.length} results
           for "<strong>{query}</strong>"
-        </p>
+        </h2>
```
Comment (ncksllvn) [inline src/applications/find-forms/containers/SearchResults.jsx:127]: Fix for navigating by headings

File: src/applications/find-forms/components/FindVaForms.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/find-forms/components/FindVaForms.jsx:31]: Heh, I've been using `sr-only` which is defined in the `packages/formation/sass/base/_va.scss` file. That one has a "newer" definition. I like it!

**Review by Mottie — APPROVED** [review]

File: src/applications/find-forms/components/FindVaForms.jsx (hunk 1 above)
Comment (ncksllvn) [reply src/applications/find-forms/components/FindVaForms.jsx:31]: I just discovered this myself!

**Pull request merged**